### PR TITLE
added maintenance test

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: utils/Maintenance.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  static switchMaintenanceModeOn(): Extrinsic {
+    return api.tx.maintenance.switchMaintenanceModeOn();
+  }
+
+  static switchUpgradabilityInMaintenanceModeOff(): Extrinsic {
+    return api.tx.maintenance.switchUpgradabilityInMaintenanceModeOff();
+  }
+
+  static witchUpgradabilityInMaintenanceModeOn(): Extrinsic {
+    return api.tx.maintenance.witchUpgradabilityInMaintenanceModeOn();
```
Comment (goncer) [inline utils/Maintenance.ts:17]: typo!

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
+/*
+ *
+ * @group sequential
```
Comment (goncer) [inline test/sequential/xyk-pallet.api.maintenance-config.test.ts:3]: let's run those tests separately in some other group, like "maintenance". I think that running this on seq could cause some issues if the maintenance mode is left enabled for some reason.

**Review by goncer — CHANGES_REQUESTED** [review]

File: utils/Maintenance.ts (hunk 1 above)
Comment (AlexChetverov) [reply utils/Maintenance.ts:17]: fixed

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/sequential/xyk-pallet.api.maintenance-config.test.ts:3]: sure, changed

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  }
+
+  const eventErrorValue = hexToU8a(
+    JSON.parse(JSON.stringify(filteredEvent[0].event.data[0])).err.module.error
+  );
+  const eventErrorIndex = JSON.parse(
+    JSON.stringify(filteredEvent[0].event.data[0])
+  ).err.module.index;
+
+  const sudoEventError = api?.registry.findMetaError({
```
Comment (goncer) [inline test/sequential/xyk-pallet.api.maintenance-config.test.ts:114]: use this beauty plz, 
findErrorMetadata
from utils.ts

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+  testUser1.addAsset(firstCurrency);
+  testUser1.addAsset(liqId);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(liqId.toNumber(), 20),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(MGA_ASSET_ID, firstCurrency, defaultPoolVolumeValue)
+    )
+  );
```
Comment (goncer) [inline test/sequential/xyk-pallet.api.maintenance-config.test.ts:135]: perhaps you can do this in the setup,

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+      )
+    ),
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+});
+
+test("maintenance- try to change Maintenance Mode with a non-foundation account THEN it failed", async () => {
```
Comment (goncer) [inline test/sequential/xyk-pallet.api.maintenance-config.test.ts:88]: can you add the following test? 
> set Maintenance mode ON
> try deactivating with sudo
> try deactivating with some other user
> set Maintenance mode OFF with fundation address

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/sequential/xyk-pallet.api.maintenance-config.test.ts]: I moved it to beforeAll.

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts (hunk 5 above)
Comment (AlexChetverov) [reply test/sequential/xyk-pallet.api.maintenance-config.test.ts:99]: Done

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOn()
+    )
+  );
+
+  const switchMaintenanceModeByUser = await Sudo.batchAsSudoFinalized(
```
Comment (goncer) [inline test/sequential/xyk-pallet.api.maintenance-config.test.ts:111]: naming convention!

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+  await checkMaintenanceStatus(true, true);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+
+  await checkMaintenanceStatus(false, false);
```
Comment (goncer) [inline test/sequential/xyk-pallet.api.maintenance-config.test.ts:180]: does this test pass? perhaps you discovered a bug!
I think that first we need to disable Upgradability and then maintenanceMode

File: test/upgrade/Upgrade.from.preRelease.test.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (goncer) [inline test/upgrade/Upgrade.from.preRelease.test.ts:1]: I know that u liked this test too much, but please remove it :P

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/sequential/xyk-pallet.api.maintenance-config.test.ts:111]: Changed

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts (hunk 7 above)
Comment (AlexChetverov) [reply test/sequential/xyk-pallet.api.maintenance-config.test.ts:180]: Upgradability doesn't work without MaintenanceMode, that's why when you deactivate MaintenanceMode, you also deactivate Upgradability

File: test/upgrade/Upgrade.from.preRelease.test.ts (hunk 8 above)
Comment (AlexChetverov) [reply test/upgrade/Upgrade.from.preRelease.test.ts:1]: Done

File: test/sequential/xyk-pallet.api.maintenance-config.test.ts (hunk 7 above)
Comment (goncer) [reply test/sequential/xyk-pallet.api.maintenance-config.test.ts:180]: you are right, it works like that :O

**Review by goncer — APPROVED** [review]

**Pull request merged**